src/codegen/templates/rust/ramble.rs.hbs

```diff
@@ -149,4 +150,38 @@ impl From<{{upper_camel pkt.name}}> for Message {
         Message::{{upper_camel pkt.name}}(value)
     }
 }
-{{/each}}
\ No newline at end of file
+{{/each}}
+
+
+// TODO: Parameterize string length to handle variable sized len field
+pub type Strlen = u16;
+
+#[derive(Debug, PartialEq, BinRead, BinWrite)]
+pub struct PString {
+    len: Strlen,
+    #[br(count = len)]
```

Comment:
This is the magic piece I was missing that allows Binread to parse the vector. This instructs the library that the field:len stores the number of bytes